packages/hathor-rpc-handler/src/rpcMethods/sendTransaction.ts

```diff
@@ -0,0 +1,188 @@
+  PinRequestResponse,
+  TriggerHandler,
+  RequestMetadata,
+  SendTransactionRpcRequest,
+  SendTransactionConfirmationPrompt,
+  SendTransactionConfirmationResponse,
+  SendTransactionLoadingTrigger,
+  SendTransactionLoadingFinishedTrigger,
+  RpcResponseTypes,
+  RpcResponse,
+  RpcMethods,
+} from '../types';
+import {
+  PromptRejectedError,
+  InvalidParamsError,
+  SendTransactionError,
+  InsufficientFundsError,
+} from '../errors';
+import { validateNetwork } from '../helpers';
+
+const sendTransactionSchema = z.object({
+  method: z.literal(RpcMethods.SendTransaction),
+  params: z.object({
+    network: z.string().min(1),
+    outputs: z.array(z.object({
+      address: z.string().optional(),
+      value: z.string().transform(val => BigInt(val)),
+      token: z.string().optional(),
+      type: z.string().optional(),
+      data: z.array(z.string()).optional(),
+    })).min(1),
+    inputs: z.array(z.object({
+      txId: z.string(),
+      index: z.number().nonnegative(),
+    })).optional(),
+    changeAddress: z.string().optional(),
+  }),
+});
+
+/**
+ * Handles the 'htr_sendTransaction' RPC request by prompting the user for confirmation
+ * and sending the transaction if confirmed.
+ * 
+ * @param rpcRequest - The RPC request object containing the transaction details.
+ * @param wallet - The Hathor wallet instance used to send the transaction.
+ * @param requestMetadata - Metadata related to the dApp that sent the RPC
+ * @param promptHandler - The function to handle prompting the user for confirmation.
+ *
+ * @returns The transaction details if successful.
+ *
+ * @throws {PromptRejectedError} If the user rejects any of the prompts.
+ * @throws {InvalidParamsError} If the request parameters are invalid.
+ * @throws {SendTransactionError} If there's an error preparing or sending the transaction.
+ * @throws {InsufficientFundsError} If there are not enough funds to complete the transaction.
+ */
+export async function sendTransaction(
+  rpcRequest: SendTransactionRpcRequest,
+  wallet: HathorWallet,
+  requestMetadata: RequestMetadata,
+  promptHandler: TriggerHandler,
+): Promise<RpcResponse> {
+  const validationResult = sendTransactionSchema.safeParse(rpcRequest);
+  
+  if (!validationResult.success) {
+    throw new InvalidParamsError(validationResult.error.errors.map(e => `${e.path.join('.')}: ${e.message}`).join(', '));
+  }
+
+  const { params } = validationResult.data;
+  validateNetwork(wallet, params.network);
+
+  // Prepare the transaction outputs
+  const sendTransactionOutputs = params.outputs.map(output => {
+    const outputData = {
+      address: output.address,
+      value: output.value,
+      token: output.token,
+      type: output.type,
+      data: output.data,
+    };
+    const typedOutput = outputData as unknown as OutputSendTransaction;
+
+    if (typedOutput.type === 'data') {
+      (typedOutput as unknown as { value: bigint }).value = BigInt(1);
+      typedOutput.token = '00';
+    }
```

Comment:
Do you need this? I think the lib method already handles it in case of a data output

Reply:
removed it.

Reply:
We actually do need it because `prepareTxData` doesn't add the 1 HTR to data outputs.